client/state/happychat/connection/actions.js

```diff
@@ -67,3 +94,265 @@ export const receiveChatTranscript = ( messages, timestamp ) => ( {
+ * as it was received from Happychat.
+ *
+ * @return { Object } Action object
+ */
+export const receiveConnect = () => ( { type: HAPPYCHAT_IO_RECEIVE_CONNECT } );
+
+/**
+ * Returns an action object for the disconnect event,
+ * as it was received from Happychat.
+ *
+ * @param { String } error The error
+ * @return { Object } Action object
+ */
+export const receiveDisconnect = error => ( {
+	type: HAPPYCHAT_IO_RECEIVE_DISCONNECT,
+	error,
+} );
+
+/**
+ * Returns an action object for the token event,
+ * as it was received from Happychat.
+ *
+ * @return { Object } Action object
+ */
+export const receiveToken = () => ( { type: HAPPYCHAT_IO_RECEIVE_TOKEN } );
+
+/**
+ * Returns an action object for the init event, as received from Happychat.
+ * Indicates that the connection is ready to be used.
+ *
+ * @param { Object } user User object received
+ * @return { Object } Action object
+ */
+export const receiveInit = user => ( { type: HAPPYCHAT_IO_RECEIVE_INIT, user } );
+
+/**
+ * Returns an action object for the unauthorized event,
+ * as it was received from Happychat
+ *
+ * @param { String } error Error reported
+ * @return { Object } Action object
+ */
+export const receiveUnauthorized = error => ( {
+	type: HAPPYCHAT_IO_RECEIVE_UNAUTHORIZED,
+	error,
+} );
+
+/**
+ * Returns an action object for the reconnecting event,
+ * as it was received from Happychat.
+ *
+ * @return { Object } Action object
+ */
+export const receiveReconnecting = () => ( { type: HAPPYCHAT_IO_RECEIVE_RECONNECTING } );
+
+/**
+ * Returns an action object for the accept event indicating the system availability,
+ * as it was received from Happychat.
+ *
+ * @param  { Object } isAvailable Whether Happychat is available
+ * @return { Object } Action object
+ */
+export const receiveAccept = isAvailable => ( {
+	type: HAPPYCHAT_IO_RECEIVE_ACCEPT,
+	isAvailable,
+} );
+
+/**
+ * Returns an action object for the message event,
+ * as it was received from Happychat.
+ *
+ * @param  { Object } message Message received
+ * @return { Object } Action object
+ */
+export const receiveMessage = message => ( { type: HAPPYCHAT_IO_RECEIVE_MESSAGE, message } );
+
+/**
+ * Returns an action object for the status event,
+ * as it was received from Happychat.
+ *
+ * @param  { String } status New chat status
+ * @return { Object } Action object
+ */
+export const receiveStatus = status => ( {
+	type: HAPPYCHAT_IO_RECEIVE_STATUS,
+	status,
+} );
+
+/**
+ * Returns an action object with the error received from Happychat
+ * upon trying to send an event.
+ *
+ * @param  { Object } error Error received
+ * @return { Object } Action object
+ */
+export const receiveError = error => ( { type: HAPPYCHAT_IO_RECEIVE_ERROR, error } );
+
+/**
+ * Returns an action object for the transcript reception.
+ *
+ * @param { Object } result An object with {messages, timestamp} props
+ * @return { Object } Action object
+ */
+export const receiveTranscript = ( { messages, timestamp } ) => ( {
```

Comment:
I think it would make it clearer what this action needs if we deconstructed the object before and send the parameters instead of an object. `export const receiveTranscript( messages, timestamp )`.

Reply:
In terms of the `receiveTranscript` method, you're right, it'd be clearer using params instead of the object.

However, the reason why I needed to use an object instead is because this method is only called by the `connection.request`([code](https://github.com/Automattic/wp-calypso/pull/19216/files#diff-c649d8554516a2203d293bf572b6352bR152)). For `connection.request` to be generic (so it can fire any kind request) the actions dispatched by in need to have the same method signature.